python-libraries/data-platform-catalogue/data_platform_catalogue/entities_v2.py

```diff
@@ -0,0 +1,262 @@
+    nullable: bool = Field(description="Whether the field is nullable or not")
+    is_primary_key: bool = Field(
+        description="Whether the field is part of the primary key"
+    )
+    foreign_keys: list[ColumnRef] = Field(
+        description="References to columns in other tables"
+    )
+
+
+class ContactRef(BaseModel):
+    """
+    A reference to a named individual that performs some kind of governance
+    """
+
+    display_name: str = Field(
+        description="The full name of the user as it should be displayed"
+    )
+    email: str = Field("Contact email for the user")
+    id: str = Field("Unique identifier for the user")
+
+
+# Needs refining
+class Governance(BaseModel):
+    """
+    Governance model for an asset or domain
+    """
+
+    data_owner: ContactRef = Field(description="")
+    data_stewards: list[ContactRef] = Field(description="")
+
+
+class DomainRef(BaseModel):
+    """
+    Reference to a domain that assets belong to
+    """
+
+    display_name: str = Field(
+        description="Display name", json_schema_extra={"example": "HMPPS"}
+    )
+    id: str = Field(
+        description="The identifier of the domain.",
+        json_schema_extra={"example": "urn:li:domain:HMCTS"},
+    )
+
+
+class TagRef(BaseModel):
+    """
+    Reference to a tag
+    """
+
+    display_name: str = Field(
+        description="Human friendly tag name", json_schema_extra={"example": "PII"}
+    )
+    id: str = Field(
+        description="The identifier of the tag",
+        json_schema_extra={"example": "urn:li:tag:PII"},
+    )
+
+
+# Needs refining
+class UsageRestrictions(BaseModel):
+    """
+    Metadata about how assets may be used.
+    """
+
+    status: DatabaseStatus = Field(
+        description="this is an enum representing the status of this version of the Data Product. Allowed values are: [draft|published|retired]. This is a metadata that communicates the overall status of the Data Product but is not reflected to the actual deployment status."
+    )
+    dpia_required: bool = Field(
+        description="Bool for if a data privacy impact assessment (dpia) is required to access this data product",
+        json_schema_extra={"example": True},
+    )
+    dpia_location: str | None = Field(description="")
+    data_sensitivity_level: SecurityClassification = Field(
+        description="", default=SecurityClassification.OFFICIAL
+    )
+
+
+# Needs refining
+class AccessInformation(BaseModel):
+    """
+    Any metadata about how to access a data asset.
+    The same data asset may be accessable via multiple means.
+    """
+
+    where_to_access_dataset: str = Field(
+        description="User-friendly description of where the data can be accessed",
+        default="",
+    )
+    source_dataset_name: str = Field(description="", default="")
+    s3_location: str | None = Field(
+        description="Location of the data in s3", default=None
+    )
+
+
+# Needs refining
+class DataSummary(BaseModel):
```

Comment:
Yeah - I think the way we talk about dates is a little confusing at the moment because we have dates the metadata was refreshed but we also have expectations about how frequently the dataset is updated that we want to communicate to the user.

I think this is tied to the data quality metrics - it's effectively a verifiable assertion about its timeliness

Reply:
Think it's the more useful of the date information vs last updated in catalogue (from a consumer perspective), but not sure it's really data quality. E.g. You might have pipelines that are run daily from live operation systems but some might only be run monthly - but that schedule and the reality of the execution of the schedule are different. So as a user of the data you'd want to know if when go to use this data when are the records contained up to

Reply:
>that schedule and the reality of the execution of the schedule are different

Hopefully this will be [addressable in the future with Data Contracts](https://datahubproject.io/docs/next/managed-datahub/observe/data-contract/) / Metadata quality. We can ingest data [from dbt's `run_results`](https://datahubproject.io/docs/next/managed-datahub/observe/data-contract/#dbt-test) or [Great Expectations](https://datahubproject.io/docs/next/managed-datahub/observe/data-contract/#great-expectations) to let us know if the scheduled ingestion / transformation ran